sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -115,7 +120,40 @@ public HttpLoggingPolicy(HttpLogDetailLevel detailLevel, boolean prettyPrintJSON
         return reqBodyLoggingMono;
     }
 
-    private Function<HttpResponse, Mono<HttpResponse>> logResponseDelegate(final ClientLogger logger, final URL url,
+    private void formatAllowableHeaders(Set<String> allowedHeaderNames, HttpRequest request, ClientLogger logger) {
```

Comment:
@samvaity / @JonathanGiles does guideline proposes different meaning for `empty vs null` value for `allowedHeaderNames`? i.e. something like empty set means redact all headers and null means log all headers?

Reply:
I think we should also add something like a "*", to say log all headers may be..

Reply:
At present null and empty set should mean the same thing - to not print any headers.